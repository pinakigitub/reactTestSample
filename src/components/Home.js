import React from 'react'

class Home extends React.Component {
    constructor(props,context) {
        super(props,context);
 
    this.state = {
       
    };
    
    }

    render(){
        return (
            <div id="sarada">Sarada</div>
        )
    }
}

export default Home;