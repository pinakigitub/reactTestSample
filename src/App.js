import React from 'react';
import Home from './components/Home'


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isDbOperationGoingOn:false
    }
  }
  render() {
    return ( 
          <div>
            <span>
            Pinaki
            </span>
             <div>
               <Home></Home>
             </div>
            </div>

    )}
};

export default App;