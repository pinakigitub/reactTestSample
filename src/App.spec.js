import React from 'react';

import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Home from './components/Home'
import App from './App';


const list = ['a', 'b', 'c'];

describe('App', () => {
  

  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
   it('Home Component  should be rendered in side App Component',()=>{
    const wrapper = mount(<App />);
    expect(wrapper.find(Home).length).toEqual(1);

   })

 
});

