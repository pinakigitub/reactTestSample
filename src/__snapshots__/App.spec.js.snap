// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <span>
    Pinaki
  </span>
  <div>
    <div>
      Sarada
    </div>
  </div>
</div>
`;
